### PR TITLE
Check 3.11 support v4

### PR DESCRIPTION
**Pull request by gaborbernat**
Signed-off-by: Bernát Gábor <gaborjbernat@gmail.com>

### CODE REVIEW

**Comment by asottile** [comment]
wouldn't be a new cpython release if cython weren't broken eh?

**Comment by gaborbernat** [comment]
> wouldn't be a new cpython release if cython weren't broken eh?

Well cython is very brittle to c-extension modifications...

**Comment by cclauss** [comment]
3.11.0-beta.5 should be available now.

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -17,40 +17,40 @@ jobs:
       fail-fast: false
       matrix:
         py:
+          - "3.11.0-beta.4"
```
Comment (cclauss) [inline .github/workflows/check.yml:20]:
```suggestion
          - "3.11.0-beta.5"
```

**Comment by fschulze** [comment]
Why is tox installing devpi-server? I did a GitHub search for it in the repository, but it didn’t come up with anything.

**Comment by cclauss** [comment]
https://github.com/tox-dev/devpi-process is written but the tox-dev team and described here https://devpi.net/docs/devpi/devpi/stable/+d/index.html

**Comment by gaborbernat** [comment]
Uses devpi in the test suite to test self provisioning.

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -17,40 +17,40 @@ jobs:
       fail-fast: false
       matrix:
         py:
+          - "3.11.0-beta.5"
```
Comment (cclauss) [inline .github/workflows/check.yml:20]:
```suggestion
          - "3.11-dev"
```

Comment (gaborbernat) [reply .github/workflows/check.yml:20]: No thanks.

Comment (cclauss) [reply .github/workflows/check.yml:20]: Why not [rc1](https://www.python.org/download/pre-releases)?

Comment (gaborbernat) [reply .github/workflows/check.yml:20]: Until aiofiles makes a new release doesn't matter. You were suggesting dev though not rc1.

Comment (cclauss) [reply .github/workflows/check.yml:20]: Understood.  `dev == rc1` until there is an `rc2` ;-)

**Comment by fschulze** [comment]
I found the tests. For now I would suggest to pin devpi-server to <6.3.0 when running with Python 3.11. These should work in pyproject.toml:
```
  "devpi-server<6.3;python_version=='3.11'",
  "devpi-server>=6.6;python_version<'3.11'",
```
There should be no changes in devpi-server between 6.2 and 6.6 which would affect the tests besides usage of asyncio.

**Pull request closed without merging**